vectordraw/vectordraw.py

```diff
@@ -0,0 +1,387 @@
+    answer = Dict(scope=Scope.user_state)
+    # Dictionary that represents result returned by the grader for the most recent answer;
+    # contains info about correctness of answer and feedback message
+    result = Dict(scope=Scope.user_state)
+
+    editable_fields = (
+        'display_name',
+        'description',
+        'width',
+        'height',
+        'bounding_box_size',
+        'axis',
+        'show_navigation',
+        'show_vector_properties',
+        'show_slope_for_lines',
+        'add_vector_label',
+        'vector_properties_label',
+        'background_url',
+        'background_width',
+        'background_height',
+        'vectors',
+        'points',
+        'expected_result',
+        'custom_checks'
+    )
+
+    has_score = True
+
+    @property
+    def settings(self):
+        """
+        Return settings for this exercise.
+        """
+        return {
+            'width': self.width,
+            'height': self.height,
+            'bounding_box_size': self.bounding_box_size,
+            'axis': self.axis,
+            'show_navigation': self.show_navigation,
+            'show_vector_properties': self.show_vector_properties,
+            'show_slope_for_lines': self.show_slope_for_lines,
+            'add_vector_label': self.add_vector_label,
+            'vector_properties_label': self.vector_properties_label,
+            'background': self.background,
+            'vectors': self.get_vectors,
+            'points': self.get_points,
+            'expected_result': self.get_expected_result
+        }
+
+    @property
+    def user_state(self):
+        """
+        Return user state, which is a combination of most recent answer and result.
+        """
+        user_state = self.answer
+        if self.result:
+            user_state['result'] = self.result
+        return user_state
+
+    @property
+    def background(self):
+        """
+        Return information about background to draw for this exercise.
+        """
+        return {
+            'src': self.background_url,
+            'width': self.background_width,
+            'height': self.background_height,
+        }
+
+    @property
+    def get_vectors(self):
+        """
+        Load info about vectors for this exercise from JSON string specified by course author.
+        """
+        return json.loads(self.vectors)
+
+    @property
+    def get_points(self):
+        """
+        Load info about points for this exercise from JSON string specified by course author.
+        """
+        points = json.loads(self.points)
+        for point in points:
+            # If author did not specify whether point should be drawn in fixed location (True)
+            # or placed by student (False), we default to True;
+            # template needs this info to determine whether it should add option
+            # for selecting point to dropdown menu:
+            if 'fixed' not in point:
+                point['fixed'] = True
```

Comment:
Since you had to move part of the default value code on the server side, it makes sense to move all of it.  We generally want to have as much of the code in Python as possible, since that code is easier to test and maintain (and also because Python is not quite as insane a language as JS).

If there are any blockers to moving the default value code here, use your own judgement to decide whether it makes sense, but if it is straight-forward, I think it woulld be an improvement.